opencv/c_extension/fast_image_processing.cpp

```diff
@@ -0,0 +1,229 @@
+#include <string>
+#include <iostream>
+#include <vector>
+#include <math.h> // atan
+#include <cmath> // std::abs
+
+#include <pybind11/pybind11.h>
+
+#include <Eigen/Core>
+#include <Eigen/Dense>
+
+#include <opencv2/core/core.hpp>
+#include <opencv2/core/eigen.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc.hpp>
+
+#define CONVERT_TO_INT true
+#define MAX_WIDTH 320
+#define PI 3.14159265
+#define REF_ANGLE float(- PI / 2)
+#define MAX_ANGLE float(PI / 4)
+#define IM_WIDTH MAX_WIDTH
+#define N_REGIONS 3
+
+#include "ndarray_converter.h"
+
+using namespace Eigen;
+using EigenMat = Eigen::Matrix<float, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>;
+
+namespace py = pybind11;
+
+bool initialized = false;
+
+
+cv::Mat centroids;
+cv::Mat resized = cv::Mat::ones(20, 80, CV_32FC3);
+
+cv::Mat w1 = cv::Mat::ones(80*20*3, 20, CV_32F).t();
```

Comment:
neural network?

Reply:
Yep, in fact, we don't really need to specify the shape, it is more for readability
w1, w2, ... are the weights, b1, b2, ... are the biases